nats.go

```diff
@@ -1174,6 +1179,22 @@ func (o Options) Connect() (*Conn, error) {
 	return nc, nil
 }
 
+func defaultErrHandler(nc *Conn, sub *Subscription, err error) {
+	var cid uint64
+	if nc != nil {
+		nc.mu.RLock()
+		cid = nc.info.CID
+		defer nc.mu.RUnlock()
```

Comment:
Is it intentional that you keep the connection's lock until the end of this function? Error handlers are already serialized, if that was the concern.

Reply:
No, will fix.